src/GPSs.cpp

```diff
@@ -0,0 +1,53 @@
+#include "GPSs.h"
+#include <Adafruit_GPS.h>
+#include <FlexCAN_T4.h>
+
+GPSs::GPSs(int id, HardwareSerial* serialIn) {
+    this->id = id;
+    this->wGPS = Adafruit_GPS(serialIn);
+}
+
+void GPSs::begin() {
+    this->wGPS.begin(9600);
+    this->wGPS.sendCommand(PMTK_API_SET_FIX_CTL_5HZ);
+    this->wGPS.sendCommand(PMTK_ENABLE_WAAS);
+}
+
+Health GPSs::healthCheck() {
+    unsigned long startCheck = millis();
+
+    while(millis() - startCheck <= 500) {
+        this->wGPS.read();
+        if(this->wGPS.newNMEAreceived()) {
+            return Health::HEALTHY;
+        }
+    }
+    return Health::UNRESPONSIVE;
+}
+
+bool GPSs::ready() {
+    this->wGPS.read();
+    return this->wGPS.newNMEAreceived();
+}
+
+SensorData GPSs::read() {
+    SensorData sensorData = SensorData(id, 1);
```

Comment:
I'd use constexpr for the number of SensorData msgs (1) and buffer length (8) so we can avoid confusion

Reply:
Gonna keep this one the way it is. More of a personal choice but I like just having the values there so I know what the len and msg count is instead of having to jump to where its defined in the file.